geomesa-hbase/geomesa-hbase-datastore/src/main/scala/org/locationtech/geomesa/hbase/data/HBaseDataStoreFactory.scala

```diff
@@ -36,39 +36,48 @@ class HBaseDataStoreFactory extends DataStoreFactorySpi with LazyLogging {
   override def createNewDataStore(params: java.util.Map[String, Serializable]): DataStore = createDataStore(params)
 
   override def createDataStore(params: java.util.Map[String, Serializable]): DataStore = {
-
     // TODO HBase Connections don't seem to be Serializable...deal with it
     val connection = HBaseConnectionPool.getConnection(params, validateConnection)
 
-    val catalog = getCatalog(params)
-
     val remoteFilters = RemoteFilteringParam.lookupOpt(params).map(_.booleanValue)
       .getOrElse(HBaseDataStoreFactory.RemoteFilterProperty.get.toBoolean)
     logger.debug(s"Using ${if (remoteFilters) "remote" else "local" } filtering")
 
-    val generateStats = GenerateStatsParam.lookup(params)
-    val audit = if (AuditQueriesParam.lookup(params)) {
+    val audit = if (!AuditQueriesParam.lookup(params)) { None } else {
       Some(AuditLogger, Option(AuditProvider.Loader.load(params)).getOrElse(NoOpAuditProvider), "hbase")
-    } else {
-      None
     }
-    val queryThreads = QueryThreadsParam.lookup(params)
-    val coprocessorThreads = CoprocessorThreadsParam.lookup(params)
-    val queryTimeout = QueryTimeoutParam.lookupOpt(params).map(_.toMillis)
-    val maxRangesPerExtendedScan = MaxRangesPerExtendedScanParam.lookup(params)
-    val maxRangesPerCoprocessorScan = MaxRangesPerCoprocessorScanParam.lookup(params)
-    val looseBBox = LooseBBoxParam.lookup(params)
-    val caching = CachingParam.lookup(params)
-    val authsProvider = if (!EnableSecurityParam.lookup(params)) { None } else {
+    val auths = if (!EnableSecurityParam.lookup(params)) { None } else {
       Some(HBaseDataStoreFactory.buildAuthsProvider(connection, params))
     }
-    val coprocessorUrl = CoprocessorUrlParam.lookupOpt(params)
-
-    val ns = NamespaceParam.lookupOpt(params)
-
-    val config = HBaseDataStoreConfig(catalog, remoteFilters, generateStats, audit, queryThreads,
-      coprocessorThreads, queryTimeout, maxRangesPerExtendedScan, maxRangesPerCoprocessorScan,
-      looseBBox, caching, authsProvider, coprocessorUrl, ns)
+    val queries = HBaseQueryConfig(
+      threads = QueryThreadsParam.lookup(params),
+      timeout = QueryTimeoutParam.lookupOpt(params).map(_.toMillis),
+      looseBBox = LooseBBoxParam.lookup(params),
+      caching = CachingParam.lookup(params),
+      maxRangesPerExtendedScan = MaxRangesPerExtendedScanParam.lookup(params)
+    )
+    val enabledCoprocessors = EnabledCoprocessors(
+      arrow = ArrowCoprocessorParam.lookup(params),
+      bin = BinCoprocessorParam.lookup(params),
+      density = DensityCoprocessorParam.lookup(params),
+      stats = StatsCoprocessorParam.lookup(params)
+    )
```

Comment:
Should we have a check to see if someone turned off remote filters and yet did something else with enabledCoprocessors?

Reply:
mmm, we could... do you think it's confusing?

Reply:
I was there for both features, so I'm clear on what's happening:).  I have no idea what other people's reaction will be.